libraries/chain/chain_controller.cpp

```diff
@@ -1304,12 +1305,10 @@ void chain_controller::update_last_irreversible_block()
       }
       if( new_producer_schedule ) {
          _db.modify( gpo, [&]( auto& props ){
-             /// TODO: use upper / lower bound to remove range
-              while( gpo.pending_active_producers.size() ) {
-                 if( gpo.pending_active_producers.front().first < new_last_irreversible_block_num ) {
-                   props.pending_active_producers.erase(props.pending_active_producers.begin());
-                 }
-              }
+              boost::range::remove_erase_if(props.pending_active_producers,
+                                            [&](typename decltype(props.pending_active_producers)::value_type const& v) -> bool {
+                                               return v.first < new_last_irreversible_block_num;
+                                            });
```

Comment:
Some style changes. Please avoid unqualified capture lists when possible. Also we use style of `const` before type not after. So this would become:

```
            boost::range::remove_erase_if(props.pending_active_producers,
                 [new_last_irreversible_block_num](const typename decltype(props.pending_active_producers)::value_type& v) -> bool {
                       return v.first < new_last_irreversible_block_num;
                 });
```